tests/test_request_history.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const { expect } = require('chai')
+const http = require('http')
+const nock = require('..')
+
+describe('Request history', () => {
+  afterEach(() => {
+    nock.cleanAll();
+  })
+
+  it('remembers requests', async () => {
+    const interceptor1 = nock('http://example.test').persist().get(/\/dsad1.*/)
```

Comment:
I'm afraid this will create OOM errors and/or "memory leak:-like behaviour.

Reply:
I'd say unit-tests typically don't send enormous payloads, or make thousands of requests. 

Anyhow: remembering 2000 get-requests getting "foo" as answer: https://gist.github.com/sezanzeb/d5b89e10bdebf2cfb6192983743aa9eb shows 101MiB of memory usage in my system monitor. Without any requests 48MiB are needed.

Also, requests should be cleaned up after every test. I just realized this still needs to be implemented, like https://jestjs.io/docs/mock-function-api#mockfnmockclear

Reply:
I agree with you, but users may use Nock in unexpected ways. WDYT about saving only the last ten requests?

Reply:
If there is a warning being written to the console when the maximum number of requests is stored, and if it is configurable, it would be an acceptable compromise.

Reply:
Sounds good.

Reply:
Alternative:

```js
nock('http://example.test')
  .persist()
  .remember(10)  // null by default: remembers all requests
  .get(/\/dsad2.*/)
```

without `.remember()`, no requests are saved.

Reply:
I think this is nice and fits well into nock, and is optional, so that it won't cause any problems for existing tests. Because it is explicitly enabled, the limit can be infinite.

Reply:
Adding features is easy. Maintaining them is something else :) 
I know that using persist is a known pattern with `nock`, but can you please share with me why you use it instead of regular interceptors?
Don't get me wrong—I'm in [favor of inserting](https://github.com/nock/nock/pull/2750#issuecomment-2318623185) this feature to `nock,` but I think we need to be careful how we implement it.

Reply:
You can also do

```js
  it('remembers a request', async () => {
    const interceptor = nock('http://example.test')
        .get('/foo')
        .remember()

    interceptor.reply(202, '1')
    await sendRequest('example.test', '/foo')
    expect(interceptor.requests).to.have.lengthOf(1);

    interceptor.reply(202, '1')
    await sendRequest('example.test', '/foo')
    expect(interceptor.requests).to.have.lengthOf(2);
  })
```

without persist.

I have commited that new test